### PR TITLE
Reimplement osrank_naive

### PR DESCRIPTION
**Pull request by MeBrei**
As `rank_network` and `random_walk` are now in a working state, `osrank_native` can be uncommented and adjusted. Note that the seed set phase is still not implemented in `random_walk`.

### CODE REVIEW

File: src/algorithm.rs [hunk 1]
```diff
@@ -16,19 +16,17 @@ use rand::Rng;
 pub enum OsrankError {}
 
 #[derive(Debug)]
-// TODO(adn) We shouldn't copy here, but rather have
-// a `network_view: &'a G`.
-pub struct WalkResult<G, I> {
-    network_view: G,
+pub struct WalkResult<'a, G, I> {
+    network_view: &'a G,
```
Comment (adinapoli-mndc) [inline src/algorithm.rs]: Ah, this is great 😉

File: src/algorithm.rs [hunk 2]
```diff
@@ -81,13 +78,32 @@ where
...
-// pub fn osrank_naive(seed_set: SeedSet, network: &mut Network) -> Result<(), OsrankError> {
-//     // Phase1, rank the network and produce a NetworkView.
-//     let phase1 = random_walk(Some(seed_set), &NetworkView::from_network(network))?;
-//     // Phase2, compute the osrank only on the NetworkView
-//     let mut phase2 = random_walk(None, &phase1.network_view)?;
-//     rank_network(&phase2.walks, &mut phase2.network_view)
-// }
+pub fn osrank_naive<L, G>(
+    seed_set: Option<SeedSet>,
+    network: &mut G,
```
Comment (adinapoli-mndc) [inline src/algorithm.rs]: This is fine for now, but perhaps something to think about for the "grand vision" -- It's conceivable that the Graph API won't give us a mutable reference to a graph, but just a read-only view, so we might have to perhaps return something that the Graph API can use to update the graph itself. This is all very hazy for now, but still work dwelling about it (cc @cloudhead ).

Comment (MeBrei) [reply src/algorithm.rs]: Yes, I was thinking the same.

Comment (cloudhead) [reply src/algorithm.rs]: I changed my mind a little on this - I think passing in a mutable graph makes sense - as it's much cheaper. What we could do though is isolate the osrank graph from the graph of balances etc.

Comment (adinapoli-mndc) [reply src/algorithm.rs]: It's cheaper for sure, but I do wonder if that's a good idea. Perhaps we will have to experiment with both strategies and see which one yields the best tradeoff in terms of performance and security 😉

**Pull request merged**